PGIP/GraphQL/Resolver/DGraph.hs

```diff
@@ -0,0 +1,75 @@
+module PGIP.GraphQL.Resolver.DGraph (resolve) where
+
+import PGIP.GraphQL.Resolver.ToResult
+
+import PGIP.GraphQL.Result as GraphQLResult
+import PGIP.GraphQL.Result.DGraph as GraphQLResultDGraph
+import PGIP.GraphQL.Result.DocumentLink as GraphQLResultDocumentLink
+import PGIP.GraphQL.Result.OMSSimple as GraphQLResultOMSSimple
+
+import PGIP.Shared
+
+import Driver.Options
+import Persistence.Database
+import Persistence.Schema as DatabaseSchema
+import Persistence.Schema.Enums as DatabaseSchemaEnums
+import Persistence.Utils
+
+import Database.Esqueleto
+
+import Control.Monad.IO.Class (MonadIO (..))
+
+resolve :: HetcatsOpts -> Cache -> String -> IO (Maybe GraphQLResult.Result)
+resolve opts sessionReference locIdVar =
+  onDatabase (databaseConfig opts) $ resolveDB locIdVar
+
+resolveDB :: MonadIO m => String -> DBMonad m (Maybe GraphQLResult.Result)
+resolveDB locIdVar = do
+  dgraphL <-
+    select $ from $ \(documents `InnerJoin` loc_id_bases) -> do
+      on (loc_id_bases ^. LocIdBaseId ==. coerceId (documents ^. DocumentId))
+      where_ (loc_id_bases ^. LocIdBaseLocId ==. val locIdVar)
+      return (documents, loc_id_bases)
+  case dgraphL of
+    [] -> return Nothing
+    (documentEntity, locIdBaseEntity@(Entity documentKey _)) : _ -> do
+      documentLinksSourceResults <- resolveDocumentLinks documentKey DocumentLinkSourceId
+      documentLinksTargetResults <- resolveDocumentLinks documentKey DocumentLinkTargetId
+      omsResults <- resolveOMS documentKey
+      case locIdBaseKind $ entityVal locIdBaseEntity of
+        DatabaseSchemaEnums.Library ->
+          return $ Just $ GraphQLResult.DGraphResult $ GraphQLResultDGraph.DGLibrary $
+            libraryToResult documentEntity locIdBaseEntity documentLinksSourceResults documentLinksTargetResults omsResults
```

Comment:
this line is 123 characters long. Please try mostly to stay within 80 character lines.

Reply:
Done in b7fb8f235